config.c

```diff
@@ -1587,7 +1587,7 @@ static int readConfigFile(const char *configFile, struct logInfo *defConfig)
 			if (newlog->oldDir) {
 				for (i = 0; i < newlog->numFiles; i++) {
 					char *ld;
-					dirName = ourDirName(newlog->files[i]);
+					dirName = dirname(strdup(newlog->files[i]));
```

Comment:
I like the idea of using the system implementation of ```dirname()``` and ```basename()``` but constructions like ```dirname(strdup(...))``` are inherently wrong.  Taking this approach you will either (continue to) leak memory or cause an invalid ```free()```.  While it is not explicitly stated in the documentation you referred to, the return values of ```dirname()``` and ```basename()``` cannot be passed to ```free()```.